frontend-desktop/src/main.rs

```diff
@@ -88,28 +92,46 @@ fn send(websocket: &Client, data: Value, event: &str) {
     websocket.emit(event, data).expect("Socket down");
 }
 
-#[derive(Debug, Deserialize)]
+#[derive(Debug, Deserialize, Serialize)]
 struct CoffeeCallerConfig {
     username: String,
     server: String,
 }
 
-fn read_config(path: &str) -> CoffeeCallerConfig {
+fn read_config(path: &str) -> Result<CoffeeCallerConfig, ConfigError> {
     let settings = Config::builder()
         .add_source(config::File::with_name(path))
         .add_source(config::Environment::with_prefix("COFFEE"))
         .build()
-        .unwrap_or_default();
+        .unwrap_or_default();    
+    return settings.try_deserialize();
+}
+
+fn create_default_config() -> CoffeeCallerConfig {
     let default_config: CoffeeCallerConfig = CoffeeCallerConfig {
         username: "RustCoffee".into(),
         server: "http://localhost:4200".into(),
     };
-    let parsed: CoffeeCallerConfig = settings.try_deserialize().unwrap_or(default_config);
-    return parsed;
+    return default_config;
+}
+
+fn disable_buttons(ui: Weak<MainWindow>) {
+    ui.unwrap().invoke_set_buttons(false);
+}
+
+fn enable_buttons(ui: Weak<MainWindow>) {
+    ui.unwrap().invoke_set_buttons(true);
 }
 
 fn main() {
-    let config = Arc::new(read_config("./config"));
+
+    let config = read_config("./config");
```

Comment:
Consider loading the file from the users config dir: https://docs.rs/dirs/1.0.4/dirs/fn.config_dir.html

Reply:
Good idea, will do